libpod/container.go

```diff
@@ -374,8 +376,32 @@ func (c *Container) Exec(cmd []string, tty bool, stdin bool) (string, error) {
 
 // Attach attaches to a container
 // Returns fully qualified URL of streaming server for the container
-func (c *Container) Attach(stdin, tty bool) (string, error) {
-	return "", ErrNotImplemented
+func (c *Container) Attach(noStdin bool, keys string) error {
+	// Check the validity of the provided keys first
+	var err error
+	detachKeys := []byte{}
+	if len(keys) > 0 {
+		detachKeys, err = term.ToBytes(keys)
+		if err != nil {
+			return errors.Wrapf(err, "invalid detach keys")
+		}
+	}
+	cStatus := c.state.State
+
+	if !(cStatus == ContainerStateRunning || cStatus == ContainerStateCreated) {
+		return errors.Errorf("%s is not created or running", c.Name())
+	}
+	resize := make(chan remotecommand.TerminalSize)
+	defer close(resize)
+	err = c.attachContainerSocket(resize, noStdin, detachKeys)
+	if err != nil {
+		return err
+	}
+	// TODO
+	// Re-enable this when mheon is done wth it
+	//c.ContainerStateToDisk(c)
```

Comment:
Can we remove the comment and have .cContainerStateToDisk just do nothing?

Reply:
I'd be more comfortable leaving it. We have plenty of TODOs in the same file.